### ISSUE TITLE
Two minor issues with the batch normalization implementation

### ISSUE THREAD

**Issue by Pandoro** [lucasb-eyer/DeepFried2#15]
The first, as discussed, you are not computing the unbiased variance estimate. This will not be ground breaking, but it might, especially for small batch sizes, make a difference and should thus for completeness just be there.

The second is an issue I see with `get_stat_updates(self)`. Here you take the stats of the last forward pass and add them to the buffer with a weight of one. This will be perfectly fine as long as you have equaly sized batches, but if you are ever going to do something else, this will be a problem. It's up to you if you want to change this or not, it was just something that didn't feel completely future-proof.

**Comment by lucasb-eyer** [comment]
Yeah, well spotted on the first one.

About the second one: In effect, that weights the last minibatch proportionately larger as it is smaller than the others. (I hope that makes sense?)
I knew about this while implementing it, and chose not to bother because, practically, it doesn't matter much: typically one iterates in minibatches and just drops the last one, which is safe to do since we shuffle between epochs.

I could fix this by also recording `symb_input[0]` in the forward pass and use that for computing the precise mean. Will give it a try sometime soon, but if it slows down a lot I might not do it and simply document this caveat.

**Comment by Pandoro** [comment]
I was more thinking about a possible future online kind of scenario, where
you might train something, but batch sizes are not always the same. I'm not
sure if this would every make sense in a practice, it's just that one
should keep this in mind if one would ever want to use it for something
like this.
